[CI] Netty4HttpServerPipeliningTests is susceptible to timeouts

From https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java10,nodes=virtual&&linux/23/console

```
20:18:30 HEARTBEAT J3 PID(3393@worker-098f258273371858e.build.us-west-2a.elasticnet.co): 2018-08-07T20:18:30, stalled for 21.5s at: Netty4HttpServerPipeliningTests.testThatHttpPipeliningCanBeDisabled
20:18:40 HEARTBEAT J3 PID(3393@worker-098f258273371858e.build.us-west-2a.elasticnet.co): 2018-08-07T20:18:40, stalled for 31.5s at: Netty4HttpServerPipeliningTests.testThatHttpPipeliningCanBeDisabled
20:18:45 Suite: org.elasticsearch.http.netty4.Netty4HttpServerPipeliningTests
...
20:18:45 FAILURE 32.8s J3 | Netty4HttpServerPipeliningTests.testThatHttpPipeliningCanBeDisabled <<< FAILURES!
20:18:45    > Throwable #1: java.lang.AssertionError: 
20:18:45    > Expected: a collection with size <6>
20:18:45    >      but: collection size was <1>
20:18:45    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
20:18:45    > 	at org.elasticsearch.http.netty4.Netty4HttpServerPipeliningTests.testThatHttpPipeliningCanBeDisabled(Netty4HttpServerPipeliningTests.java:150)
```

Based on the 30+ second stalling, and the resulting failure it looks like it timed-out waiting on this latch:
https://github.com/elastic/elasticsearch/blob/c313f11e1f06b7b84bbbe2c70e69fca976c15a28/modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpClient.java#L148

It might be better to have an explicitly timeout failure rather than continue on and fail with an unrelated assertion. (Or just increase the timeout).

--- a/modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpClient.java
+++ b/modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpClient.java
@@ -58,6 +58,7 @@ import java.util.concurrent.TimeUnit;
 
 import static io.netty.handler.codec.http.HttpHeaderNames.HOST;
 import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static org.junit.Assert.fail;
 
 /**
  * Tiny helper to send http requests over netty.
@@ -145,7 +146,9 @@ class Netty4HttpClient implements Closeable {
             for (HttpRequest request : requests) {
                 channelFuture.channel().writeAndFlush(request);
             }
-            latch.await(30, TimeUnit.SECONDS);
+            if (latch.await(30L, TimeUnit.SECONDS) == false) {
+                fail("Failed to get all expected responses.");
+            }
 
         } finally {
             if (channelFuture != null) {

--- a/plugins/transport-nio/src/test/java/org/elasticsearch/http/nio/NioHttpClient.java
+++ b/plugins/transport-nio/src/test/java/org/elasticsearch/http/nio/NioHttpClient.java
@@ -71,6 +71,7 @@ import java.util.function.Consumer;
 import static io.netty.handler.codec.http.HttpHeaderNames.HOST;
 import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
 import static org.elasticsearch.common.util.concurrent.EsExecutors.daemonThreadFactory;
+import static org.junit.Assert.fail;
 
 /**
  * Tiny helper to send http requests over nio.
@@ -136,7 +137,9 @@ class NioHttpClient implements Closeable {
             for (HttpRequest request : requests) {
                 nioSocketChannel.getContext().sendMessage(request, (v, e) -> {});
             }
-            latch.await(30, TimeUnit.SECONDS);
+            if (latch.await(30L, TimeUnit.SECONDS) == false) {
+                fail("Failed to get all expected responses.");
+            }
 
         } catch (IOException e) {
             throw new UncheckedIOException(e);

